### PR TITLE
Additional timestamp field for wheel and body velocity

### PR DESCRIPTION
**Pull request by bkolligs**
Adding lower level timestamps, and changed data type of the time structure. 

### New timestamp fields
When an encoder message is generated, it goes through the following events:
1. The encoder itself is queried by the MSP
2. The peripheral computer receives data from the MSP
3. The peripheral computer sends data to the central computer
4. The central computer receives data from the peripheral computer
5. The peripheral interface publishes encoder data to the cFS software bus, potentially converting ticks to velocities
6. The vehicle controller receives raw wheel speeds, and converts it to body velocity
7. The pose estimator receives body velocity 

With this in mind, we need to use the closest time to the encoders to calculate the rover's local displacement. I've added fields that hold the timestamp from event 1 or 2 (whenever we choose to stamp the encoder data) to the higher level body velocity and wheel velocity messages. This lets us use the most accurate time difference between encoder ticks when calculating velocity. 

### Data type change
Changed the actual data type of the timestamp from `CFE_Time_SysTime_t` to `OS_time_t`, because the former is not guaranteed to be monotonically increasing, while the latter is. 

## 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have read and followed the `CONTRIBUTING.md` file
- [x] I have applied the .clang-format file to my changes
- [x] I have commented my code using Doxygen style comments (see Contributing MD for examples), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design documentation 
- [ ] I have updated any relevant parameters in the internal ICD
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this is a bug-fix, I have added tests that can catch similar bugs in the future
- [ ] New and existing unit tests pass locally with my changes
- [ ] For cfe apps I have reserved an empty address for my msgs_ids and perf_ids from the [tracking list](https://docs.google.com/spreadsheets/d/1yRUhNTRBOiGLswlsWftNaYlwGKDeNQ_fCnDYFrc_7fU/edit#gid=772812381) and checked that it does not clash with other apps
- [ ] For new added message types, there is a timestamp included via either an explicit timestamp variable in the struct or buried in the a tlmHeader.

### CODE REVIEW

File: fsw/public_inc/body_velocity_msg.h [hunk 1]
```diff
@@ -27,30 +28,30 @@ typedef float float32;
  * Type definition (MOONRANGER body velocity packet)
  */
 typedef struct {
-  CFE_TIME_SysTime_t timeStamp;   // nanoseconds
-  float32            xVelocity;   // m/s
-  float32            yawVelocity; // rads/s
+    OS_time_t timeStamp;                // time that this message was generated
+    OS_time_t sourceEncoderTimestamp;   // timestamp belonging to the encoder
```
Comment (bkolligs) [inline fsw/public_inc/body_velocity_msg.h:32]: This may be unncessary

Comment (bkolligs) [reply fsw/public_inc/body_velocity_msg.h:32]: This is neccessary

**Review by tenzincrouch — COMMENTED** [review]
We need to sort out data types. I thought we had moved to int16?

**Comment by bkolligs** [comment]
We did, though I think we were talking about the wheel packet sent by the peripheral computer to the peripheral interface. Should I update these central computer side messages to use int16 for the telemetry as well?

File: fsw/public_inc/wheel_velocity_current_msg.h [hunk 2]
```diff
@@ -15,38 +15,46 @@
...
-    float32 leftBack;     // RPM
-    float32 rightBack;    // RPM
+    OS_time_t sourceEncoderTimestamp;   // timestamp belonging to the encoder
+                                        // data used to generate this message
+    /* The left front motor speed in RPM*/
+    int16 motorOne;
+    /* The right front motor speed in RPM */
+    int16 motorTwo;
+    /* The left back motor speed in RPM */
+    int16 motorThree;
```
Comment (bkolligs) [inline fsw/public_inc/wheel_velocity_current_msg.h:35]: @tenzincrouch this is ready for your review

**Review by bdcmu — APPROVED** [review]

**Review by bdcmu — APPROVED** [review]

**Pull request merged**